projects/packages/jitm/src/class-jitm.php

```diff
@@ -80,18 +80,38 @@ public function register() {
 	 * @param \WP_Screen $screen WP Core's screen object.
 	 */
 	public function prepare_jitms( $screen ) {
-		if ( ! in_array(
-			$screen->id,
-			array(
-				'jetpack_page_akismet-key-config',
-				'admin_page_jetpack_modules',
-			),
-			true
-		) ) {
-			add_action( 'admin_enqueue_scripts', array( $this, 'jitm_enqueue_files' ) );
-			add_action( 'admin_notices', array( $this, 'ajax_message' ) );
-			add_action( 'edit_form_top', array( $this, 'ajax_message' ) );
+		// Disable all JITMs on these pages.
+		if (
+			in_array(
+				$screen->id,
+				array(
+					'jetpack_page_akismet-key-config',
+					'admin_page_jetpack_modules',
+				),
+				true
+			) ) {
+			return;
+		}
+
+		// Disable all JITMs on pages where the recommendations banner is displaying.
+		if (
+			in_array(
+				$screen->id,
+				array(
+					'dashboard',
+					'plugins',
+					'jetpack_page_stats',
+				),
+				true
+			)
+			&& \Jetpack_Recommendations_Banner::can_be_displayed()
```

Comment:
I wonder if this may be problematic, as we're trying to have the JITM package be more independent from Jetpack. @kbrown9 Can you think of a different way to do this?

Reply:
Yes, this will be problematic. Thanks for looking out for the independence of the packages!

@robertf4 - Our goal is to get the packages into a state where they can be used by other plugins without Jetpack being installed. So, we need to make sure that we don’t use any classes/functions from the Jetpack plugin in the packages.

I don’t see an existing way for plugins to control the JITMs based on the screen id. Maybe we could add a new filter in the `JITM::prepare_jitms()` method, and that filter would pass along the screen id. The logic for determining whether to display a JITM when the Recommendations Banner is displayed could be located in the Jetpack plugin instead of in the JITM package.

For example, we could add something like this top at the top of `JITM::prepare_jitms()`:

```
		if ( ! apply_filters( ‘jetpack_display_jitms’, true, $screen->id ) ) {
			return;
		}
```

Then, we could hook onto the `jetpack_display_jitms` filter in the `Jetpack_Recommendations_Banner` class.

What do you think of this approach?

Reply:
I like that idea! The only thing I changed was to hook onto `jetpack_display_jitms` in `class.jetpack-admin.php` since I also moved the filtering for the akismet key config and modules pages as well. That decouples the JITM package from any reference to a Jetpack admin page and keeps all the filtering logic for the plugin in one place.